homeassistant/components/sensor/__init__.py

```diff
@@ -546,6 +546,87 @@ class SensorStateClass(StrEnum):
+    SensorDeviceClass.CURRENT: {None, SensorStateClass.MEASUREMENT},
+    SensorDeviceClass.DATA_RATE: {None, SensorStateClass.MEASUREMENT},
+    SensorDeviceClass.DATA_SIZE: {
+        None,
+        SensorStateClass.TOTAL,
+        SensorStateClass.TOTAL_INCREASING,
+    },
+    SensorDeviceClass.DATE: {None},
+    SensorDeviceClass.DISTANCE: {None}.union(set(SensorStateClass)),
+    SensorDeviceClass.DURATION: {None},
+    SensorDeviceClass.ENERGY: {
+        None,
+        SensorStateClass.TOTAL,
+        SensorStateClass.TOTAL_INCREASING,
+    },
+    SensorDeviceClass.ENUM: {None},
+    SensorDeviceClass.FREQUENCY: {None, SensorStateClass.MEASUREMENT},
+    SensorDeviceClass.GAS: {
+        None,
+        SensorStateClass.TOTAL,
+        SensorStateClass.TOTAL_INCREASING,
+    },
+    SensorDeviceClass.HUMIDITY: {None, SensorStateClass.MEASUREMENT},
+    SensorDeviceClass.ILLUMINANCE: {None, SensorStateClass.MEASUREMENT},
+    SensorDeviceClass.IRRADIANCE: {None, SensorStateClass.MEASUREMENT},
+    SensorDeviceClass.MOISTURE: {None, SensorStateClass.MEASUREMENT},
+    SensorDeviceClass.MONETARY: {
+        None,
+        SensorStateClass.TOTAL,
+        SensorStateClass.TOTAL_INCREASING,
+    },
+    SensorDeviceClass.NITROGEN_DIOXIDE: {None, SensorStateClass.MEASUREMENT},
+    SensorDeviceClass.NITROGEN_MONOXIDE: {None, SensorStateClass.MEASUREMENT},
+    SensorDeviceClass.NITROUS_OXIDE: {None, SensorStateClass.MEASUREMENT},
+    SensorDeviceClass.OZONE: {None, SensorStateClass.MEASUREMENT},
+    SensorDeviceClass.PM1: {None, SensorStateClass.MEASUREMENT},
+    SensorDeviceClass.PM10: {None, SensorStateClass.MEASUREMENT},
+    SensorDeviceClass.PM25: {None, SensorStateClass.MEASUREMENT},
+    SensorDeviceClass.POWER_FACTOR: {None, SensorStateClass.MEASUREMENT},
+    SensorDeviceClass.POWER: {None, SensorStateClass.MEASUREMENT},
+    SensorDeviceClass.PRECIPITATION: {
+        None,
+        SensorStateClass.TOTAL,
+        SensorStateClass.TOTAL_INCREASING,
+    },
+    SensorDeviceClass.PRECIPITATION_INTENSITY: {None, SensorStateClass.MEASUREMENT},
+    SensorDeviceClass.PRESSURE: {None, SensorStateClass.MEASUREMENT},
+    SensorDeviceClass.REACTIVE_POWER: {None, SensorStateClass.MEASUREMENT},
+    SensorDeviceClass.SIGNAL_STRENGTH: {None, SensorStateClass.MEASUREMENT},
+    SensorDeviceClass.SOUND_PRESSURE: {None, SensorStateClass.MEASUREMENT},
+    SensorDeviceClass.SPEED: {None, SensorStateClass.MEASUREMENT},
+    SensorDeviceClass.SULPHUR_DIOXIDE: {None, SensorStateClass.MEASUREMENT},
+    SensorDeviceClass.TEMPERATURE: {None, SensorStateClass.MEASUREMENT},
+    SensorDeviceClass.TIMESTAMP: {None},
+    SensorDeviceClass.VOLATILE_ORGANIC_COMPOUNDS: {None, SensorStateClass.MEASUREMENT},
+    SensorDeviceClass.VOLTAGE: {None, SensorStateClass.MEASUREMENT},
+    SensorDeviceClass.VOLUME: {
+        None,
+        SensorStateClass.TOTAL,
+        SensorStateClass.TOTAL_INCREASING,
+    },
```

Comment:
Probably here also, as it's a generic volume it can be fluctuating (quantity of petrol left in a tank tends to go down).

Reply:
> quantity of petrol left in a tank tends to go down
So it is a total?

Reply:
Total increasing could maybe make sense if this is used for meters other than gas and water meters?

Reply:
Yeah, you could essentially add a meter to any fluid (at least, that is how I approached it).